testr/packages.py

```diff
@@ -96,6 +95,35 @@ def fileno(self):
         return sys.stdout.fileno()
 
 
+def communicate(process, log=None, text=False):
+    """
+    Real-time reading of a subprocess stdout.
+    Parameters
+    ----------
+    process:
+        process returned by subprocess.Popen
+    log: a stream object
+    level: str or int
+        the logging level
+    text: bool
+        whether the process output is text or binary
+    """
+
+    while True:
+        if process.poll() is not None:
+            break
+        line = process.stdout.readline()
+        line = line if text else line.decode()
+        log.write(line)
```

Comment:
Need to check if `log is not None`.

Reply:
that, or make `log` a required argument. In the code I copied this from, I was doing something like
```
if log is None:
    log = logging
```